src/journal-remote/journal-upload.c

```diff
@@ -121,6 +131,83 @@ static int check_cursor_updating(Uploader *u) {
         return 0;
 }
 
+int config_parse_compression(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        char *compression = ASSERT_PTR(data);
+        if (isempty(rvalue)) {
+                *compression = COMPRESSION_NONE;
+                return 0;
+        }
+
+        _cleanup_free_ char *s = strdup(rvalue);
+        if (!s)
+                return log_oom();
+
+        Compression c = compression_from_string(ascii_strupper(s));
+        if (c < 0)
+                return log_syntax_parse_error(unit, filename, line, c, lvalue, rvalue);
+        if (!compression_supported(c))
+                return log_syntax(unit, LOG_WARNING, filename, line, 0,
+                           "Compression=%s is not supported on a system", rvalue);
+
+        *compression = c;
+        return 0;
+}
+
+int config_parse_header(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        char ***headers = ASSERT_PTR(data);
+        char *unescaped;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+
+        if (isempty(rvalue)) {
+                /* Empty assignment resets the list */
+                *headers = strv_free(*headers);
+                return 0;
+        }
+
+        if (!header_is_valid(rvalue)) {
```

Comment:
if this is an escaped string you should unescape first, and then validate it as a header